cmd/root.go

```diff
@@ -162,7 +163,7 @@ func rootCmd(o *options.Options) error {
 	// intentionally placed at end to preserve outputting results, even if a check has a runtime error
 	for _, result := range repoResult.Checks {
 		if result.Error != nil {
-			return fmt.Errorf("one or more checks had a runtime error: %w", result.Error)
+			return sce.WithMessage(sce.ErrorCheckRuntime, fmt.Sprintf("%s: %v", result.Name, result.Error))
```

Comment:
do we need to make this a list of errors, since multiple checks may have failed?

Reply:
It depends on what we want. For the purposes of `scorecard-action` the error is just getting matched with `errors.Is` so the details don't matter.
It also wasn't necessary in #2133 since the info for all errors were visible if you ran with `--show-details`. The main point was returning a singular error so the status code would be 1 when any checks had failures.

Reply:
Thanks for the clarification. I suppose the problem is if there are multiple checks failing, each with a different error. Is it a concern?

Reply:
From a behavior point of view: no. As long as there is one error, the CLI exit code is non-zero and `scorecard-action` can ignore the errors. The code is at the very end of `rootCmd` so it's purely information for status code.

From an informational point of view, collating multiple errors would benefit CLI users that don't have `--show-details`, but that's about it. The error info still makes it to `scorecard-action`. For example, if you look at the [current badge for scorecard](https://api.securityscorecards.dev/projects/github.com/ossf/scorecard), you'll see the info already there:  `{"name":"Branch-Protection","score":-1,"reason":"internal error: error during branchesHandler.setup: .......`

Reply:
One limitation is that scorecard action does not have access to a structured version of the results, just to a file. Correct? So I'm wondering if it's a problem if the Action filters based on the single error, even though there may be multiple distinct errors.
I agree it's fine for a human who can read the results.

Reply:
Ignore my comment: action will only filter for `ErrorCheckRuntime`, so not a problem.